owncloudData/src/test/java/com/owncloud/android/data/transfers/implementation/OCLocalTransferDataSourceTest.kt

```diff
@@ -0,0 +1,326 @@
+        val timestamp = System.currentTimeMillis()
+
+        ocLocalTransferDataSource.updateTransferWhenFinished(id, TransferStatus.TRANSFER_SUCCEEDED, timestamp, TransferResult.UPLOADED)
+
+        verify(exactly = 1) {
+            transferDao.updateTransferWhenFinished(id, TransferStatus.TRANSFER_SUCCEEDED.value, timestamp, TransferResult.UPLOADED.value)
+        }
+    }
+
+    @Test
+    fun `updateTransferLocalPath changes transfer local path correctly` () {
+
+        ocLocalTransferDataSource.updateTransferLocalPath(id, ocTransfer.localPath)
+
+        verify(exactly = 1) {
+            transferDao.updateTransferLocalPath(id, ocTransfer.localPath)
+        }
+    }
+
+    @Test
+    fun `updateTransferStorageDirectoryInLocalPath changes directory correctly`() {
+        val oldDirectory = "oldDirectory"
+        val newDirectory = "newDirectory"
+
+        ocLocalTransferDataSource.updateTransferStorageDirectoryInLocalPath(id, oldDirectory, newDirectory)
+
+        verify(exactly = 1) {
+            transferDao.updateTransferStorageDirectoryInLocalPath(id, oldDirectory, newDirectory)
+        }
+    }
+
+    @Test
+    fun `deleteTransferById removes it correctly`() {
+
+        ocLocalTransferDataSource.deleteTransferById(id)
+
+        verify(exactly = 1) {
+            transferDao.deleteTransferWithId(id)
+        }
+    }
+
+    @Test
+    fun `deleteAllTransfersFromAccount removes all transfers correctly`() {
+
+        ocLocalTransferDataSource.deleteAllTransfersFromAccount(OC_ACCOUNT_NAME)
+
+        verify(exactly = 1) {
+            transferDao.deleteTransfersWithAccountName(OC_ACCOUNT_NAME)
+        }
+    }
+
+    @Test
+    fun `getTransferById returns a OCTransfer`() {
+        every {
+            transferDao.getTransferWithId(any())
+        } returns ocTransfer.toEntity()
+
+        val actualResult = ocLocalTransferDataSource.getTransferById(id)
+
+        assertEquals(ocTransfer, actualResult)
+
+        verify(exactly = 1) {
+            transferDao.getTransferWithId(id)
+        }
+    }
+
+    @Test
+    fun `getAllTransfers returns a list of OCTransfer`() {
+
+        every {
+            transferDao.getAllTransfers()
+        } returns listOf(transferEntity)
+
+        val actualResult = ocLocalTransferDataSource.getAllTransfers()
+
+        assertEquals(listOf(transferEntity.toModel()), actualResult)
+
+        verify(exactly = 1) {
+            transferDao.getAllTransfers()
+        }
+    }
+
+    @Test
+    fun `getAllTransfersAsStream returns a flow of list of OCTransfer ordered by status`() = runBlocking {
+
+        val transferEntityInProgress: OCTransferEntity = ocTransfer.toEntity().copy(status = 0)
+
+        val transferEntityQueue: OCTransferEntity = ocTransfer.toEntity().copy(status = 1)
+
+        val transferEntityFailed: OCTransferEntity = ocTransfer.toEntity().copy(status = 2)
+
+        val transferEntitySucceeded: OCTransferEntity = ocTransfer.toEntity().copy(status = 3)
```

Comment:
Setting the `status` with a raw number (`= 1`, `= 2`...) in the entities is not correct. Which status is 1? Or which status is 2? We must use the real type which, at the same time, is more understandable. Check the `OCTransfer` implementation and set the status in the model class instead of the entity, so that when we do a `toEntity()` the corresponding number is already set

Reply:
Done